source/extensions/filters/http/credential_injector/config.cc

```diff
@@ -0,0 +1,50 @@
+#include "source/extensions/filters/http/credential_injector/config.h"
+
+#include "source/common/config/utility.h"
+#include "source/extensions/filters/http/credential_injector/credential_injector_filter.h"
+#include "source/extensions/injected_credentials/common/factory.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace CredentialInjector {
+
+using Envoy::Extensions::InjectedCredentials::Common::NamedCredentialInjectorConfigFactory;
+
+absl::StatusOr<Http::FilterFactoryCb>
+CredentialInjectorFilterFactory::createFilterFactoryFromProtoTyped(
+    const envoy::extensions::filters::http::credential_injector::v3::CredentialInjector&
+        proto_config,
+    const std::string& stats_prefix, Server::Configuration::FactoryContext& context) {
+  // Find the credential injector factory.
+  const std::string type{
+      TypeUtil::typeUrlToDescriptorFullName(proto_config.credential().typed_config().type_url())};
+  NamedCredentialInjectorConfigFactory* const config_factory =
+      Registry::FactoryRegistry<NamedCredentialInjectorConfigFactory>::getFactoryByType(type);
+  if (config_factory == nullptr) {
+    return absl::InvalidArgumentError(
+        fmt::format("Didn't find a registered implementation for type: '{}'", type));
+  }
```

Comment:
Please use `Config::Utility::getFactory< NamedCredentialInjectorConfigFactory>(proto_config.credential())` here. It will handle the `TypedStruct` case automatically and may downgrade to use config name if necessary.